nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,844 @@
+            batch_size=cfg.batch_size,
+            shuffle=cfg.shuffle,
+            num_workers=cfg.num_workers,
+            pin_memory=cfg.pin_memory,
+            drop_last=cfg.drop_last,
+            collate_fn=dataset.collate_fn,
+        )
+
+    def _setup_infer_dataloader(self, queries: List[str], test_ds) -> 'torch.utils.data.DataLoader':
+        """
+        Setup function for a infer data loader.
+        Args:
+            queries: text
+            batch_size: batch size to use during inference
+        Returns:
+            A pytorch DataLoader.
+        """
+
+        dataset = IntentSlotInferenceDataset(
+            tokenizer=self.tokenizer, queries=queries, max_seq_length=-1, do_lower_case=False
+        )
+
+        return torch.utils.data.DataLoader(
+            dataset=dataset,
+            collate_fn=dataset.collate_fn,
+            batch_size=test_ds.batch_size,
+            shuffle=test_ds.shuffle,
+            num_workers=test_ds.num_workers,
+            pin_memory=test_ds.pin_memory,
+            drop_last=test_ds.drop_last,
+        )
+
+    def update_data_dirs(self, data_dir: str, dialogues_example_dir: str):
+        """
+        Update data directories
+
+        Args:
+            data_dir: path to data directory
+            dialogues_example_dir: path to preprocessed dialogues example directory, if not exists will be created.
+        """
+        if not os.path.exists(data_dir):
+            raise ValueError(f"{data_dir} is not found")
+        self.cfg.dataset.data_dir = data_dir
+        self.cfg.dataset.dialogues_example_dir = dialogues_example_dir
+        logging.info(f'Setting model.dataset.data_dir to {data_dir}.')
+        logging.info(f'Setting model.dataset.dialogues_example_dir to {dialogues_example_dir}.')
+
+    def predict_from_examples(self, queries: List[str], test_ds) -> List[List[str]]:
+        """
+        Get prediction for the queries (intent and slots)
+        Args:
+            queries: text sequences
+            test_ds: Dataset configuration section.
+        Returns:
+            predicted_intents, predicted_slots: model intent and slot predictions
+        """
+
+        predicted_intents = []
+        predicted_slots = []
+        mode = self.training
+
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+        # Switch model to evaluation mode
+        self.eval()
+        self.to(device)
+
+        # Dataset.
+        infer_datalayer = self._setup_infer_dataloader(queries, test_ds)
+
+        for batch in infer_datalayer:
+            input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask = batch
+
+            intent_logits, slot_logits = self.forward(
+                input_ids=input_ids.to(device),
+                token_type_ids=input_type_ids.to(device),
+                attention_mask=input_mask.to(device),
+            )
+
+            # predict intents
+            intent_preds = tensor2list(torch.argmax(intent_logits, axis=-1))
```

Comment:
There seem to be many things from the validation_step that can be used here. Can you create a new method that contains these shared things to be called in both the `validation_step` and here